### DISCUSSION TITLE
A way to know which Puppet versions are supported

### DISCUSSION THREAD

**Discussion by jay7x** [puppetlabs/pdk#1327] [Feature Request]
Sometimes it might be useful to know which Puppet versions are supported in this particular PDK release. For example, if one would like to build a test matrix based on this. So having some CLI option or a sub-command would be great. PDK may emit a JSON with the lowest and highest Puppet versions supported (and/or maybe Facter & Ruby versions also)

**Comment by davidsandilands** [comment]
Do you feel it would have to be a command or would a callable well formatted part of release notes / documentation work here?

**Reply by jay7x** [reply]
Ideally I'd prefer it to be a command.. imagine a Github Action, for example. One would like to run a `pdk test unit --puppet-version=<value>` for every supported value.. but how to know those supported values? One can hardcode the list but then it should be updated with every new PDK release. That's why I'd like to have a way to know the supported versions list.
